---
talk-title: " Cooperative learning for multiview analysis"
author: "Tony Liang"
course: "BMEG524"
talk-date: "March 18, 2024"
format: revealjs
---

<!--- This looks very ugly TODO: FIX THIS --->

<br>
<br>

::: {.primary .bold .justify-center .f-subheadline .tj .flex}

{{< meta talk-title  >}}

:::

<br>

<br>


::: {.f2 .bold}

{{< meta author >}} 


{{< meta course >}}


{{< meta talk-date >}}

::: 



## Multiomics data

Different types of molecular measurements (omics) from common set of samples

- Genomics
- Proteomics
- Metabolomics
- And more


## Types of fusion strategy

- Early

  - Merge multiple data views into single representation than feed to model

- Late
  
  - Individual models fit on distinct view, then predictions of thes models are combined into final predictor

- Mixing of both

Analogy to LASSO, Ridge or elastic-net

## Paper

The paper is from @ding2022cooperative, supervised learning algorithm that focus on disease classification and treatment response prediction

::: {.fragment}

- Addresses "fusion" in data-adaptive manner
- Combines different omics together, they call this "view"

:::


## Motivation and background 1

Takes in data from various modalities mentioned earlier, integrating heterogenous features on common set of observations (say patients)

::: {.callout-note}

# Example

Input from same patients:

- RNA expression
- DNA-methylation measurements

Predict cancer outcomes

:::

::: {.fragment}

1. Both data views potentially have prognostic value
2. Two view share some underlysing relationship with each other

:::

::: {.notes}

- DNA methylation regulates gene expression --> could repress expression tumor suppresor genes  or promote expression of oncogenes

:::


## When would you want to do this analysis

- Multiple modalities from same biological samples exist, specially when they share some underlying relationship in their signals

- Making discoveries that are hidden in data analyses of single modality

- Achieving more accurate predictions of the outcome variable


## Motivation and background 3

::: {.callout-note}


Using this algorithm, we could model outcome of interest better by fusing different data views

:::


## Algorithm 1
<!-- 
::: {.callout-note}
- What is the theoretical underpinnings of the approach
::: -->

- A 
- B
- C

## Algorithm 2

::: {.callout-note}
- How does it work?
:::

## Algorithm 3

- Requires omics from same providers
- Runtime complexity is high due to iterative nature of the algorithm
- Works best if the omics itself are related each other



## Quality control measures 1

::: {.callout-note}
- How can you determine whether it worked?
:::

## What happens if data quality is not good?

When the views are not correlated, this could be very [bad]{.secondary}^[Appendix Fig S3 from @ding2022cooperative]

```{r}
#| echo: false
#| fig-cap: "Both X, Z contains signal, no correlation"
knitr::include_graphics("assets/app_s3c-a.png")
```

## What happens if data quality is not good?

When the views are not correlated, this could be very [bad]{.secondary}^[Appendix Fig S3 from @ding2022cooperative]

```{r}
#| echo: false
#| layout-ncol: 2
#| fig-cap: 
#| - "Both X, Z contains signal, no correlation"
#| - "Only X contains signal"
knitr::include_graphics("assets/app_s3c-a.png")
knitr::include_graphics("assets/app_s3c-c.png")
```

## What happens if data quality is not good?

When the views are not correlated, this could be very [bad]{.secondary}^[Appendix Fig S3 from @ding2022cooperative]

```{r}
#| echo: false
#| layout-ncol: 3
#| fig-cap: 
#| - "Both X, Z contains signal"
#| - "X with more signal than Z"
#| - "Only X contains signal"
knitr::include_graphics("assets/app_s3c-a.png")
knitr::include_graphics("assets/app_s3c-b.png")
knitr::include_graphics("assets/app_s3c-c.png")
```

## Analysis 1

::: {.callout-note}
- Are there special analysis considerations?
:::

## Analysis 2

::: {.callout-note}
- Controlling for sources of bias
:::

## Analysis 3

::: {.callout-note}
- Insights into analysis?
:::

## Analysis 4

::: {.callout-note}
- What is the biological interpretation?
:::


## {#end}

<br>
<br>

<br>
<br>

::: {.closing}

Questions?

:::



## {#reference}